packages/node/src/message-handling/handle-protocol-message.ts

```diff
@@ -36,138 +37,213 @@ export async function handleReceivedProtocolMessage(
     router
   } = requestHandler;
 
-  const deserializedMsg = bigNumberifyJson(msg);
-
-  const { data } = deserializedMsg;
+  const { data } = bigNumberifyJson(msg) as NodeMessageWrappedProtocolMessage;
```

Comment:
Based on offline discussion we should avoid type casting like this.

Reply:
I have to type case in this case because `bigNumberifyJson` destroys type information.